src/lib.rs

```diff
@@ -102,14 +102,62 @@ impl Default for Options {
 /// *Errors* are only happening if the underlying buffer fails, which is unlikely.
 pub fn cmark_with_options<'a, I, E, F>(
     events: I,
-    mut formatter: F,
+    formatter: F,
     state: Option<State<'static>>,
     options: Options,
 ) -> Result<State<'static>, fmt::Error>
 where
     I: Iterator<Item = E>,
     E: Borrow<Event<'a>>,
     F: fmt::Write,
+{
+    cmark_with_options_and_buffer(events.map(|e| (None, e)), formatter, state, options)
+}
+
+/// As `cmark_with_options`, but with default `Options`.
+pub fn cmark<'a, I, E, F>(
+    events: I,
+    formatter: F,
+    state: Option<State<'static>>,
+) -> Result<State<'static>, fmt::Error>
+where
+    I: Iterator<Item = E>,
+    E: Borrow<Event<'a>>,
+    F: fmt::Write,
+{
+    cmark_with_options(events, formatter, state, Options::default())
+}
+
+pub fn into_buffer_offset_iter<'a>(
+    input: &'a str,
```

Comment:
Here is where the 'split brain' could occour, as `input` must match the one used to create `parser`. Unfortunately as of now there is no way to obtain the `text` from the [`Parser`](https://docs.rs/pulldown-cmark/0.8.0/pulldown_cmark/struct.Parser.html), which might very well be a contribution of our own making.